src/qibolab/channel.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass, field
+
+from qibo.config import raise_error
+
+from .channel_configs import ChannelConfig
+
+
+@dataclass(frozen=True)
```

Comment:
I made it frozen so that the users are not able to do things like this:
```python
ch = platform.get_qubit(0).flux
ch.config.offset = 0.3
```

Instead, the correct way to manipulate the said channel variable, is to provide it as execution argument, which will update it from its value in platform parameters only during the execution.

Reply:
`frozen` means that the data can not be reassigned, but not that all the data accessible from are immutable:
```py
[ins] In [1]: from dataclasses import dataclass
                                                                                                                                                                                               
[ins] In [2]: @dataclass
         ...: class A:
         ...:     a: int
         ...:
                                                                                                                                                                                               
[ins] In [3]: @dataclass(frozen=True)
         ...: class B:
         ...:     b: A
         ...:
                                                                                                                                                                                               
[ins] In [4]: c = B(A(3)); c
Out[4]: B(b=A(a=3))

[ins] In [5]: c.b = A(4)
---------------------------------------------------------------------------
FrozenInstanceError                       Traceback (most recent call last)
Cell In[5], line 1
----> 1 b.b = A(4)

File <string>:4, in __setattr__(self, name, value)

FrozenInstanceError: cannot assign to field 'b'

[ins] In [6]: c.b.a = 4; c.b.a
Out[6]: 4
```